Update oras-go to v1.1.1

macox opened the pull request:
Update oras-go to the latest major stable version. This version contained breaking changes.

Fixes #682 

Required by https://github.com/open-policy-agent/conftest/pull/698

jalseth commented on internal/commands/push.go:
```diff
@@ -107,45 +105,49 @@ func NewPushCommand(ctx context.Context, logger *log.Logger) *cobra.Command {
 
-	memoryStore := content.NewMemoryStore()
-	layers, err := buildLayers(ctx, memoryStore, path)
+	registry, err := content.NewRegistry(content.RegistryOptions{PlainHTTP: false})
```
Does specifying `PlainHTTP: false` here do anything? I suspect not as booleans in Go default to `false`.

jalseth commented on internal/commands/push.go:
```diff
@@ -107,45 +105,49 @@ func NewPushCommand(ctx context.Context, logger *log.Logger) *cobra.Command {
 }
 
-func buildLayers(ctx context.Context, memoryStore *content.Memorystore, path string) ([]ocispec.Descriptor, error) {
+func buildLayers(ctx context.Context, memoryStore *content.Memory, path string) (*content.Memory, []ocispec.Descriptor, error) {
```
Why are we modifying the function signature here?

jalseth reviewed (commented):
Hi, thank you for the PR! I left some comments but overall the changes look reasonable. Once they have been addressed I will test manually since we don't have any automated acceptance tests for this functionality.

macox commented:
> Hi, thank you for the PR! I left some comments but overall the changes look reasonable. Once they have been addressed I will test manually since we don't have any automated acceptance tests for this functionality.

Thanks @jalseth, I think have addressed your comments.
I would really appreciate the manual test 🙂

jalseth commented:
This change does not appear to work, error below. I tested the same command with v0.31.0 and it worked without issue.

```
$ ./conftest push -p examples/kubernetes/policy us-west1-docker.pkg.dev/test-gar-348023/conftest01   
2022/04/22 16:33:57 pushing bundle to: us-west1-docker.pkg.dev/test-gar-348023/conftest01:latest
Error: push bundle: pushing manifest: unknown reference: us-west1-docker.pkg.dev/test-gar-348023/conftest01:latest
```

Please review the changes and test yourself once you have resolved the issue. Google Cloud Artifact Registry has a free tier if you do not already have a registry to test with.

macox commented:
> Please review the changes and test yourself once you have resolved the issue. Google Cloud Artifact Registry has a free tier if you do not already have a registry to test with.

@jalseth finally got around to sorting this, and I have tested the push and pull commands using an AWS ECR.

I also amended the change to bump to the lastet stable version of oras-go v1.1.1 (rather than v1.0.0)

jalseth commented on downloader/oci_getter.go:
```diff
@@ -31,23 +33,29 @@ func (g *OCIGetter) Get(path string, u *url.URL) error {
+
+	registry := content.Registry{Resolver: resolver}
 	if err != nil {
-		return fmt.Errorf("new resolver: %w", err)
+		return fmt.Errorf("creating registry: %w", err)
 	}
```
It doesn't look like the previous line's assignment for `registry` has an error value, so this is reusing the same error from the `cli.ResolverWithOpts()` call above. I think this can just be removed.

jalseth commented on internal/commands/push.go:
```diff
@@ -107,44 +108,70 @@ func NewPushCommand(ctx context.Context, logger *log.Logger) *cobra.Command {
-	extraOpts := []oras.PushOpt{oras.WithConfigMediaType(openPolicyAgentConfigMediaType)}
-	manifest, err := oras.Push(ctx, resolver, repository, memoryStore, layers, extraOpts...)
+	registry := content.Registry{Resolver: resolver}
+	if err != nil {
+		return nil, fmt.Errorf("creating registry: %w", err)
+	}
```
Same comment as in `pull.go`, this error check is reusing the same error from `buildLayers()` above since the `registry` assignment does not return an error. I think this can also be removed.

jalseth reviewed (approved):
Thanks! Seems to work as expected. Just a couple small nits, and then I think we can merge.

macox commented:
Thanks for your reviews, much appreciated.
